src/components/screens/HistoryScreen/mappers/payment.tsx

```diff
@@ -0,0 +1,167 @@
+        code: destAssetCode,
+        type: assetType,
+        issuer: {
+          key: assetIssuer,
+        },
+      }}
+      size="lg"
+    />
+  );
+
+  const ActionIconComponent = isRecipient ? (
+    <Icon.ArrowCircleDown size={16} color={themeColors.foreground.primary} />
+  ) : (
+    <Icon.ArrowCircleUp size={16} color={themeColors.foreground.primary} />
+  );
+
+  const transactionTitle = `${isRecipient ? t("history.transactionHistory.received") : t("history.transactionHistory.sent")} ${destAssetCode}`;
+
+  const transactionDetails: TransactionDetails = {
+    operation,
+    transactionTitle,
+    transactionType: TransactionType.PAYMENT,
+    status: TransactionStatus.SUCCESS,
+    fee,
+    IconComponent,
+    ActionIconComponent,
+    externalUrl: `${stellarExpertUrl}/op/${id}`,
+    paymentDetails: {
+      assetCode: destAssetCode,
+      assetIssuer: assetIssuer || "",
+      assetType,
+      amount,
+      from,
+      to,
+    },
+  };
+
+  return {
+    transactionDetails,
+    rowText: destAssetCode,
+    actionText: isRecipient
+      ? t("history.transactionHistory.received")
+      : t("history.transactionHistory.sent"),
+    dateText: date,
+    amountText: formattedAmount,
+    IconComponent,
+    isAddingFunds: isRecipient,
+    ActionIconComponent,
+    transactionStatus: TransactionStatus.SUCCESS,
+  };
+};
+
+/**
+ * Renders payment transaction details
+ */
+export const PaymentTransactionDetailsContent: React.FC<{
+  transactionDetails: TransactionDetails;
+}> = ({ transactionDetails }) => {
+  const { themeColors } = useColors();
+
+  return (
+    <TransactionDetailsContent>
+      <View className="flex-row justify-between">
+        <View>
+          <Text xl primary medium numberOfLines={1}>
+            {transactionDetails.paymentDetails?.amount}{" "}
+            {transactionDetails.paymentDetails?.assetCode}
+          </Text>
+          <Text md secondary numberOfLines={1}>
+            {/* TODO: add rate */}-
+          </Text>
+        </View>
+        <AssetIcon
+          token={{
+            code: transactionDetails.paymentDetails?.assetCode ?? "",
+            issuer: {
+              key: transactionDetails.paymentDetails?.assetIssuer ?? "",
+            },
+            type: transactionDetails.paymentDetails
+              ?.assetType as AssetTypeWithCustomToken,
+          }}
+        />
+      </View>
+
+      <Icon.ChevronDownDouble
+        size={20}
+        color={themeColors.foreground.primary}
+        circle
+        circleBackground={themeColors.background.tertiary}
+      />
+
+      <View className="flex-row justify-between">
+        <View>
+          <Text xl primary medium numberOfLines={1}>
+            {truncatePublicKey({
+              publicKey: transactionDetails.paymentDetails?.to ?? "",
+              length: 4,
+            })}
+          </Text>
+          <Text md secondary numberOfLines={1}>
+            {t("history.transactionDetails.firstTimeSend")}
```

Comment:
we should remove this while we don't have the "firstTimeSend" logic in place. I think it's fine to remove since it's low priority.